enhancements/network/allow-from-router-networkpolicy.md

```diff
@@ -0,0 +1,236 @@
+`LoadBalancerService` endpoint publishing strategy, the router pods
+run on the pod network, and so router traffic can be allowed by
+creating NetworkPolicies that allow traffic based on an appropriate
+`namespaceSelector`. We currently label the `openshift-ingress`
+namespace with the label `network.openshift.io/policy-group: ingress`
+for exactly this purpose.
+
+However, when the router is run with the `HostNetwork` endpoint
+publishing strategy, the router pods will be on the host network, and
+so their traffic will not be recognized as coming from any specific
+namespace.
+
+In OpenShift SDN, for somewhat accidental historical reasons, traffic
+coming from any non-pod source is treated as though it came from the
+namespace "`default`". Thus, it's possible to allow traffic from
+routers by adding a label to `default` and then allowing traffic based
+on that label. (This will also allow traffic from any other
+host-network pod, or node process.) However, this approach does not
+work with any other network plugin.
+
+In theory, with other plugins, if you knew the pod network CIDR, you
+could create a policy saying "allow from anywhere (that can route to
+the pod network) except the pod network", which is essentially
+equivalent to "allow from `default`" in OpenShift SDN. eg:
+
+    ipBlock:
+      cidr: 0.0.0.0/0
+      except:
+        - 10.128.0.0/14
+
+This is not ideal, because it requires you to know the pod network
+CIDR, but even given that, it would not actually work with many
+network plugins anyway. Eg, in ovn-kubernetes, when a node or a
+host-network pod sends traffic to a pod-network pod, the source IP is
+not the node's "primary" node IP but rather the IP associated with the
+`ovn-k8s-mp0` interface, which is inside the pod network CIDR (eg, it
+would be `10.128.6.2` on the node that has the `10.128.6.0/23`
+subnet).
+
+It would be possible to create a policy allowing specifically the
+relevant source node IP of each node, but such a policy would need to
+be updated any time nodes were added or removed from the cluster.
+
+(There has been some discussion upstream about allowing [`nodeSelector`
+in NetworkPolicies] but this is not going to happen any time soon, if
+it happens at all.)
+
+[`nodeSelector` in NetworkPolicies]: https://github.com/kubernetes/kubernetes/issues/51891
+
+### Plan
+
+There does not seem to be any good way to solve this completely
+network-plugin-agnostically at the current time, so we won't worry
+about Kuryr, Calico, etc, at this time. (Though Kuryr could choose to
+implement the same hack I propose for OVN Kubernetes.)
+
+We also don't want to diverge too much from upstream/"stock"
+NetworkPolicy, so trying to add `nodeSelector` is out.
+
+The simplest approach would be to formalize OpenShift SDN's ability to
+"allow from host-network", and add a similar behavior to OVN
+Kubernetes. I believe this could be implemented by having
+ovn-kubernetes add the management port IP and "primary" node IP of
```

Comment:
this will entail the CNO creating and applying the namespace manifest and also ovn-k adding special code for handling this namespace based on a parameter passed to it in the "KubernetesConfig" section?

Reply:
Handling the namespace implies adding the mp0 IPs and potentially the primary node IPs to the ingress and egress address sets for that namespace.

Reply:
yes